sys/kern/thread.c

```diff
@@ -131,7 +131,7 @@ void thread_delete(thread_t *td) {
   pool_free(P_THREAD, td);
 }
 
-thread_t *thread_self(void) {
+__no_sanitize thread_t *thread_self(void) {
   return PCPU_GET(curthread);
```

Comment:
Please consider adding a short comment explaining why `__no_sanitize` is required here.